### DISCUSSION TITLE
Move GitCommands to a separate repository

### DISCUSSION THREAD

**Discussion by miloszkukla** [gitextensions/gitextensions#10241] [Ideas] [answered]
According to my quick look-around, GitCommands directory is the most extensive `git.exe` wrapper available in the .NET open-source space. It would be beneficial for .NET ecosystem to be able to use it outside of GitExtensions as an alternative to libgit2. The simplest way would be to move it to a separate repository within gitextensions organization and reference it as a submodule. If enough interest from a community would emerge a nuspec and other library niceties of its own could be added in the future but the idea is to not be an additional weight for GitExtensions project.

Additional question would be the licensing - currently, according to LICENSE.md it is GPL3 - I'm not sure what is the conclusion of https://github.com/gitextensions/gitextensions/issues/4737 though as it says in the end: 

> We received a sign off and release from a significant number of past and present contributors under MIT license. There are still few past contributors who haven't provided their sign off, but by far and large we have moved off GPL.

If it would be possible to re-license it as MIT it would be super cool but even as GPL3 I believe it would be a valuable library.

**Comment by gerhardol** [comment, marked as answer]
The relicensing has not moved forward, so GPL is the current license.

The dependencies to the application is not strictly separate, so it would not be simple to separate GitCommands to a project outside GE. It still requires Git.
I doubt the core team will drive such a change, but if it seem to be usable, it could be accepted as a contribution.
